### DISCUSSION TITLE
How can I set a specific Slint component in Rust, and could you provide an example? Thank you very much.

### DISCUSSION THREAD

**Discussion by daydaystudylab** [slint-ui/slint#6732] [Q&A] [answered]
How can I set a specific Slint component in Rust, and could you provide an example? Thank you very much🙏

**Comment by laycookie** [comment]
The same way you would define a slint component in the `.slint` file.
Here is a working example:
```rs
fn main() {
    let ui = Test::new().unwrap();

    ui.run().unwrap();
}

slint::slint!(
    import { VerticalBox, LineEdit } from "std-widgets.slint";

    export component Test inherits Window {
        VerticalBox {
            max-width: 768px;
            Text {
                text: "test";
            }
            LineEdit {
                height: 50px;
                placeholder-text: "Token";
            }
        }
    }
);
```

**Reply by daydaystudylab** [reply]
If I want to set up a LineEdit, how should I do it, thank you.

**Reply by laycookie** [reply, marked as answer]
@daydaystudylab What do you mean by "set up LineEdit"?

**Reply by daydaystudylab** [reply]
Set properties for LineEdit.

**Reply by hunger** [reply]
Slint only exposes a limited set of components to the business logic: Rule of thumb: Those that are exported by the top level file. And there you get setters and getters (as needed) for the properties and callbacks in the top-level element -- based on visibility.

So you will need to pass the properties down to child elements when needed.

Sometimes it makes sense to export a global object with properties/callbacks and such. That is then also exported and you can set properties and register callbacks via that global, too, in addition to the top level component. You can then use that global in all slint code (you might need to import it of course;-).

I like putting all the interface between business logic and UI into a global with a clear name. That makes it very clear which changes require communication between the UI teams and the backend teams: Only those in that global;-)

To me this is in no way a limitation, it is the thing I like best in slint: It enforces a clear separation of UI and business logic, it allows us to optimize your UI a lot (we know nobody uses some rust code to meddle with the objects behind our backs) and it allows us to support multiple languages for the business logic without too much effort.

**Reply by hunger** [reply]
https://github.com/slint-ui/slint/discussions/6733 has an example of how to use both approaches.

**Reply by daydaystudylab** [reply]
Thank you very much
